src/app/pages/camera.cpp

```diff
@@ -291,59 +497,174 @@ QWidget *CameraPage::network_cam_selector()
+            if(caps != nullptr)
+            {
+                GstVideoInfo* vinfo = gst_video_info_new();
+                gst_video_info_from_caps(vinfo, caps);
+                DASH_LOG(info) << "[CameraPage] Video Width: " << vinfo->width;
+                DASH_LOG(info) << "[CameraPage] Video Height: " << vinfo->height;
+            }
+
+            return GST_PAD_PROBE_REMOVE;
+        }
+    }
+
+    return GST_PAD_PROBE_OK;
+}
+
+void CameraPage::disconnect_stream()
+{
+    DASH_LOG(info) << "[CameraPage] Disconnecting camera and destroying gstreamer pipeline";
+    GstElement* capsFilter = gst_bin_get_by_name(GST_BIN(vidPipeline_), "mycapsfilter");
+    GstPad* convertPad = gst_element_get_static_pad(capsFilter, "sink");
+    gst_pad_add_probe(convertPad, GST_PAD_PROBE_TYPE_EVENT_DOWNSTREAM, &CameraPage::convertProbe, this, nullptr);
+    gst_element_set_state(vidPipeline_, GST_STATE_NULL);
+    g_object_unref(vidPipeline_);
+    if (this->config->get_cam_autoconnect()) {
+            qDebug() << "Camera disconnected. Auto reconnect in" << this->config->get_cam_autoconnect_time_secs() << "seconds";
+            this->reconnect_message = this->status->text() + ". reconnecting in %1 ";
+            this->reconnect_in_secs = this->config->get_cam_autoconnect_time_secs();
+            this->reconnect_timer->start(1000);
+    }
+    this->connected=false;
 }
 
 void CameraPage::connect_local_stream()
 {
+    this->videoContainer_ = this->local_video_widget;
     if (this->local_cam != nullptr) {
         delete this->local_cam;
         this->local_cam = nullptr;
     }
-
     const QString &local = this->config->get_cam_local_device();
     if (!this->local_cam_available(local)) {
         this->status->setText("Camera unavailable");
         return;
     }
-
-    qDebug() << "Connecting to local cam " << local;
     this->local_cam = new QCamera(local.toUtf8(), this);
-    this->local_cam->setViewfinder(this->local_video_widget);
-    connect(this->local_cam, &QCamera::statusChanged, this, &CameraPage::update_local_status);
-    this->local_cam->start();
+    this->local_cam->load();
+    qDebug() << "Camera status: " << this->local_cam->status();
+
+    QSize res = this->choose_video_resolution();
+
+
+
+    DASH_LOG(info) << "[CameraPage] Creating GStreamer pipeline with "<<this->config->get_cam_local_device().toStdString();
+    std::string pipeline = "v4l2src device="+this->config->get_cam_local_device().toStdString() + 
+                           " ! capsfilter caps=\"video/x-raw,width="+std::to_string(res.width())+",height="+std::to_string(res.height())+";image/jpeg,width="+std::to_string(res.width())+",height="+std::to_string(res.height())+"\"" +
+                           " ! decodebin";
+    init_gstreamer_pipeline(pipeline);
+    //emit the connected signal before we resize anything, so that videoContainer has had time to resize to the proper dimensions
+    emit connected_local();
+    if(videoContainer_ == nullptr)
```

Comment:
looks like this is duplicated? Might be worth refactoring it out into its own function?